### PR TITLE
REST practice

### PR DESCRIPTION
**Pull request by boazco**

### CODE REVIEW

File: src/main/kotlin/com/hibob/academy/resource/ownerResource.kt [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import jakarta.ws.rs.core.MediaType
+import org.springframework.web.bind.annotation.RequestBody
+
+@Controller
+@Path("api/owners")
+@Produces(MediaType.APPLICATION_JSON)
+class OwnerResource {
+
+    @POST
+    @Path("/{Id}/type")
```
Comment (brainpicker) [inline src/main/kotlin/com/hibob/academy/resource/ownerResource.kt]: I think you might have forgot to fix this route. When creating a pet you don't usually decide on the ID.

File: src/main/kotlin/com/hibob/academy/resource/ownerResource.kt [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    @Consumes(MediaType.APPLICATION_JSON)
+    fun createPet(@RequestBody pets: Pets): Response {
+        Response.status(Response.Status.OK).build()
```
Comment (brainpicker) [inline src/main/kotlin/com/hibob/academy/resource/ownerResource.kt]: You have in each function 2 rows of Response....status..build. Only the last will be returned. FYI

File: src/main/kotlin/com/hibob/academy/resource/ownerResource.kt [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    @Path("/{Id}/type")
+    @Consumes(MediaType.APPLICATION_JSON)
+    fun updatePetType(@PathParam("Id") id: String, petType: String): Response {
+        Response.status(Response.Status.OK).build()
+        return Response.ok("PUT OK").build()
+    }
+
+    @DELETE
+    @Path("/{Id}/type")
+    fun deletePetType(@PathParam("Id") id: String): Response {
```
Comment (brainpicker) [inline src/main/kotlin/com/hibob/academy/resource/ownerResource.kt]: What does delete pet type do actually? Maybe you meant delete pet?

File: src/main/kotlin/com/hibob/academy/resource/ownerResource.kt [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    fun getPetType(@PathParam("Id") id: String): Response {
+        Response.status(Response.Status.OK).build()
+        return Response.ok("GET OK").build()
+    }
+
+    @PUT
+    @Path("/{Id}/type")
+    @Consumes(MediaType.APPLICATION_JSON)
+    fun updatePetType(@PathParam("Id") id: String, petType: String): Response {
+        Response.status(Response.Status.OK).build()
```
Comment (brainpicker) [inline src/main/kotlin/com/hibob/academy/resource/ownerResource.kt]: For the sake of learning as much as possible, try to fake (mock) a case where you get a pet id that doesn't exist. What would you do in that case?

**Review by brainpicker — COMMENTED** [review]
Hey, I put some minor comments, but then I saw that there is a lot of code copy/pasted that wasn't fixed. All the owners resource function are talking about pets instead of owners. Also the data classes have mis-matching styles (`date_of_arrival` vs. `dateOfArrival`) and (`LocalDateTime` vs `Date`) - please go over it again and fix it before I review it again.

**Comment by boazco** [comment]
Hey, sorry for the mess, I added function for practice and fix the comments

File: src/main/kotlin/com/hibob/academy/resource/ownerResource.kt [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    @Consumes(MediaType.APPLICATION_JSON)
+    fun updateOwnersPetType(@PathParam("Id") id: String, ownersPetType: String): Response {
+        if(fetchOwner(id) == null){
+            return Response.status(404).build()
+        }
+        return Response.ok("PUT OK").build()
+    }
+
+    @DELETE
+    @Path("/{Id}/type")
```
Comment (brainpicker) [inline src/main/kotlin/com/hibob/academy/resource/ownerResource.kt]: You still missed fixing this path. What does type do in this route?

File: src/main/kotlin/com/hibob/academy/resource/petsResource.kt [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+import org.springframework.web.bind.annotation.RequestBody
+
+@Controller
+@Path("api/pets")
+@Produces(MediaType.APPLICATION_JSON)
+class PetsResource {
+
+    @POST
+    @Path("/type")
+    @Consumes(MediaType.APPLICATION_JSON)
```
Comment (brainpicker) [inline src/main/kotlin/com/hibob/academy/resource/petsResource.kt:18]: What does type do here as well? You're not creating a new type here. You're creating a new pet. It's confusing.

File: src/main/kotlin/com/hibob/academy/resource/petsResource.kt [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+    fun updatePetType(@PathParam("petId") id: String, petType: String): Response {
+        if(fetchPet(id) == null) {
+            return Response.status(404).build()
+        }
+        return Response.ok("PUT OK").build()
+    }
+
+    @DELETE
+    @Path("/{petId}/type")
```
Comment (brainpicker) [inline src/main/kotlin/com/hibob/academy/resource/petsResource.kt]: Same here with the type.

File: src/main/kotlin/com/hibob/academy/types/types.kt [hunk 8]
```diff
@@ -0,0 +1,21 @@
+package com.hibob.academy.types
+
+import java.time.LocalDateTime
+import java.util.UUID
+
+data class Pets(
+    val id: UUID = UUID.randomUUID(),
+    var name: String,
+    var type: String,
+    var company_id: UUID,
```
Comment (brainpicker) [inline src/main/kotlin/com/hibob/academy/types/types.kt]: Still mismatching styles.

**Review by brainpicker — APPROVED** [review]

**Pull request merged**